imports/builder_unix.go

```diff
@@ -101,12 +99,22 @@ func (b *Builder) Instantiate(ctx context.Context, runtime wazero.Runtime) (ctxr
 
 	for _, stdio := range []struct {
 		fd   int
+		open int
 		path string
 	}{
-		{stdin, "/dev/stdin"},
-		{stdout, "/dev/stdout"},
-		{stderr, "/dev/stderr"},
+		{stdin, syscall.O_RDONLY, "/dev/stdin"},
+		{stdout, syscall.O_WRONLY, "/dev/stdout"},
+		{stderr, syscall.O_WRONLY, "/dev/stderr"},
 	} {
+		var err error
+		if stdio.fd < 0 {
+			stdio.fd, err = syscall.Open(stdio.path, stdio.open, 0)
```

Comment:
I changed this so the module never implicitly shares the underlying files used by the host fds.